File: blobstore/sample code/golang-new/golang/oss/sample_oss.go
package oss

import (
	"context"
	"fmt"
	"github.com/minio/minio-go/v7"
	"github.com/minio/minio-go/v7/pkg/credentials"
	"io"
	"log"
	"simple_code/golang/properties"
)

type Bucket struct {
	client *minio.Client
}

//use minio-go
func NewOSSClient(param *properties.Credentials) (b Bucket, err error) {
	credential := credentials.NewStaticV4(param.AccessKey, param.Secretkey, "")
	b.client, err = minio.New(param.Endpoint, &minio.Options{
		Creds: credential,
		Secure: param.SSL,
	})
	if err != nil {
		log.Println(err)
	}
	return b, err
}

func (b *Bucket) ListObject(bucketName string) (objects []string) {
	var results <-chan minio.ObjectInfo
	results = b.client.ListObjects(context.Background(), bucketName, minio.ListObjectsOptions{})
	for result := range results {
		objects = append(objects, result.Key)
	}
	return objects
}

func (b *Bucket) UploadObject(bucketName string, objectName string, reader io.Reader) error {
	rsp, err := b.client.PutObject(context.Background(), bucketName, objectName, reader, -1, minio.PutObjectOptions{})
	if err != nil {
		//HandleError(err)
	}
	fmt.Println(rsp, "--------------------------")
	return err
}

func (b *Bucket) DownloadObject(bucketName string, objectName string) (object *minio.Object, err error) {
	object, err = b.client.GetObject(context.Background(), bucketName, objectName, minio.GetObjectOptions{})
	if err != nil {
		//HandleError(err)
	}
	return object, err
}

func (b *Bucket) DeleteObject(bucketName string, objectName string) error {
	err := b.client.RemoveObject(context.Background(), bucketName, objectName, minio.RemoveObjectOptions{})
	if err != nil {
		//HandleError(err)
	}
	return err
}

File: blobstore/sample code/python/sample_oss.py
import oss2


class OSS:
    client = None

    def __init__(self, endpoint, access_key, secret_key, bucket_name):
        self.client = oss2.Bucket(oss2.Auth(access_key_id=access_key, access_key_secret=secret_key),
                                  endpoint=endpoint,
                                  bucket_name=bucket_name)

    # -------list objects---------------
    def list_object(self):
        list = []
        objects = self.client.list_objects()
        for obj in objects.object_list:
            list.append(obj.key)
        return list

    # -------upload object---------------
    def upload_object(self, object_name, str):
        self.client.put_object(object_name, str)
        print("upload object:", object_name)

    # -------download object---------------
    def download_object(self, object_name):
        object = self.client.get_object(object_name)
        return object.read()

    # -------delete object---------------
    def delete_object(self, object_name):
        self.client.delete_object(object_name)
        print("delete object:", object_name)
